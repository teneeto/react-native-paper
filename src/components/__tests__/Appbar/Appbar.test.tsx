import React from 'react';
import { Animated, Platform } from 'react-native';

import { render } from '@testing-library/react-native';
import mockSafeAreaContext from 'react-native-safe-area-context/jest/mock';
import renderer from 'react-test-renderer';

import Provider from '../../../core/Provider';
import { getTheme } from '../../../core/theming';
import overlay from '../../../styles/overlay';
import { tokens } from '../../../styles/themes/v3/tokens';
import Appbar from '../../Appbar';
import AppbarAction from '../../Appbar/AppbarAction';
import AppbarBackAction from '../../Appbar/AppbarBackAction';
import AppbarContent from '../../Appbar/AppbarContent';
import AppbarHeader from '../../Appbar/AppbarHeader';
import {
  getAppbarColor,
  modeTextVariant,
  renderAppbarContent as utilRenderAppbarContent,
} from '../../Appbar/utils';
import Menu from '../../Menu/Menu';
import Searchbar from '../../Searchbar';
import Tooltip from '../../Tooltip/Tooltip';
import Text from '../../Typography/Text';

jest.mock('react-native-safe-area-context', () => mockSafeAreaContext);

const renderAppbarContent = utilRenderAppbarContent as (
  props: Parameters<typeof utilRenderAppbarContent>[0]
) => { props: any }[];

describe('Appbar', () => {
  it('does not pass any additional props to Searchbar', () => {
    const tree = renderer
      .create(
        <Appbar>
          <Searchbar placeholder="Search" value="" />
        </Appbar>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });

  it('passes additional props to AppbarBackAction, AppbarContent and AppbarAction', () => {
    const tree = renderer
      .create(
        <Appbar>
          <Appbar.BackAction onPress={() => {}} />
          <Appbar.Content title="Examples" />
          <Appbar.Action icon="menu" onPress={() => {}} />
        </Appbar>
      )
      .toJSON();

    expect(tree).toMatchSnapshot();
  });
});

describe('renderAppbarContent', () => {
  const children = [
    <Appbar.BackAction onPress={() => {}} key={0} />,
    <Appbar.Content title="Examples" key={1} />,
    <Appbar.Action icon="magnify" onPress={() => {}} key={2} />,
    <Appbar.Action icon="menu" onPress={() => {}} key={3} />,
  ];

  it('should render all children types if renderOnly is not specified', () => {
    const result = renderAppbarContent({
      isV3: false,
      children,
      isDark: false,
    });

    expect(result).toHaveLength(4);
  });

  it('should render all children types except specified in renderExcept', () => {
    const result = renderAppbarContent({
      isV3: false,
      children: [
        ...children,
        <Menu
          key={4}
          anchor={<Appbar.Action icon="menu" onPress={() => {}} />}
          visible={false}
        >
          {null}
        </Menu>,
      ],
      isDark: false,
      renderExcept: [Appbar, AppbarHeader, AppbarBackAction, AppbarContent],
    });

    expect(result).toHaveLength(3);
  });

  it('should render only children types specifed in renderOnly', () => {
    const result = renderAppbarContent({
      isV3: false,
      children,
      isDark: false,
      renderOnly: [AppbarAction],
    });

    expect(result).toHaveLength(2);
  });

  it('should render AppbarContent with correct mode', () => {
    const result = renderAppbarContent({
      isV3: false,
      children,
      isDark: false,
      renderOnly: [AppbarContent],
      mode: 'large',
    });

    expect(result[0].props.mode).toBe('large');
  });

  it('should render centered AppbarContent', () => {
    const renderResult = (isV3 = true) =>
      renderAppbarContent({
        children,
        isDark: false,
        isV3,
        renderOnly: [AppbarContent],
        mode: 'center-aligned',
        shouldCenterContent: true,
      });

    const centerAlignedContent = {
      alignItems: 'center',
    };

    expect(renderResult()[0].props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(centerAlignedContent)])
    );

    expect(renderResult(false)[0].props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(centerAlignedContent)])
    );
  });

  it('should not render centered AppbarContent for Android, if not V3', () => {
    Platform.OS = 'android';
    const renderResult = (isV3 = true) =>
      renderAppbarContent({
        children,
        isDark: false,
        isV3,
        renderOnly: [AppbarContent],
        mode: 'center-aligned',
        shouldCenterContent: !isV3 && Platform.OS === 'ios',
      });

    const centerAlignedContent = {
      alignItems: 'center',
    };

    expect(renderResult(false)[0].props.style).not.toEqual(
      expect.arrayContaining([expect.objectContaining(centerAlignedContent)])
    );
  });

  it('should render centered AppbarContent always for iOS, if not V3', () => {
    Platform.OS = 'ios';
    const renderResult = (isV3 = true) =>
      renderAppbarContent({
        children,
        isDark: false,
        isV3,
        renderOnly: [AppbarContent],
        mode: 'center-aligned',
        shouldCenterContent: !isV3 && Platform.OS === 'ios',
      });

    const centerAlignedContent = {
      alignItems: 'center',
    };

    expect(renderResult(false)[0].props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(centerAlignedContent)])
    );
  });

  it('should render AppbarContent with correct spacings', () => {
    const renderResult = (isV3 = true, withAppbarBackAction = false) =>
      renderAppbarContent({
        children,
        isDark: false,
        isV3,
        renderOnly: [AppbarContent, withAppbarBackAction && AppbarBackAction],
      });

    const v2Spacing = {
      marginLeft: 8,
    };

    const v3Spacing = {
      marginLeft: 12,
    };

    expect(renderResult()[0].props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(v3Spacing)])
    );

    expect(renderResult(false, true)[1].props.style).toEqual(
      expect.arrayContaining([expect.objectContaining(v2Spacing)])
    );
  });

  it('Is recognized as a header when no onPress callback has been pressed', () => {
    const { getByRole } = render(
      <mockSafeAreaContext.SafeAreaProvider>
        <Appbar.Header>
          <Appbar.Content title="Accessible test" />
        </Appbar.Header>
      </mockSafeAreaContext.SafeAreaProvider>
    );

    expect(getByRole('text')).toBeTruthy();
  });
  it('Is recognized as a button when onPress callback has been pressed', () => {
    const { getByRole } = render(
      <mockSafeAreaContext.SafeAreaProvider>
        <Appbar.Header>
          <Appbar.Content title="Accessible test" onPress={() => {}} />
        </Appbar.Header>
      </mockSafeAreaContext.SafeAreaProvider>
    );

    expect(getByRole('button')).toBeTruthy();
  });
});

describe('AppbarAction', () => {
  it('should be rendered with default theme color', () => {
    const { getByTestId } = render(
      <Appbar>
        <Appbar.Action icon="menu" testID="appbar-action" />
      </Appbar>
    );
    const appbarActionIcon = getByTestId('appbar-action').props.children[0];
    expect(appbarActionIcon.props.color).toBe(
      getTheme().colors.onSurfaceVariant
    );
  });

  it('should be rendered with specific theme color if is leading', () => {
    const { getByTestId } = render(
      <Appbar>
        <Appbar.Action icon="menu" testID="appbar-action" isLeading />
      </Appbar>
    );
    const appbarActionIcon = getByTestId('appbar-action').props.children[0];
    expect(appbarActionIcon.props.color).toBe(getTheme().colors.onSurface);
  });

  it('should be rendered with custom color', () => {
    const { getByTestId } = render(
      <Appbar>
        <Appbar.Action icon="menu" color="purple" testID="appbar-action" />
      </Appbar>
    );
    const appbarActionIcon = getByTestId('appbar-action').props.children[0];
    expect(appbarActionIcon.props.color).toBe('purple');
  });

  it('should render AppbarBackAction with custom color', () => {
    const { getByTestId } = render(
      <Appbar>
        <Appbar.BackAction color="purple" testID="appbar-action" />
      </Appbar>
    );
    const appbarBackActionIcon = getByTestId('appbar-action').props.children[0];
    expect(appbarBackActionIcon.props.color).toBe('purple');
  });

  describe('When V2', () => {
    const theme = { isV3: false };

    it('should be rendered with the right color when no color is passed', () => {
      const { getByTestId } = render(
        <Appbar theme={theme}>
          <Appbar.Action icon="menu" testID="appbar-action" />
        </Appbar>
      );

      const appbarActionIcon = getByTestId('appbar-action').props.children[0];

      expect(appbarActionIcon.props.color).toBe('#ffffff');
    });

    it('should be rendered with the right color when no color is passed but is wrapped by a Tooltip', () => {
      const { getByTestId } = render(
        <Provider>
          <Appbar theme={theme}>
            <Tooltip title="Menu">
              <Appbar.Action icon="menu" testID="appbar-action" />
            </Tooltip>
          </Appbar>
        </Provider>
      );

      const appbarActionIcon = getByTestId('appbar-action').props.children[0];

      expect(appbarActionIcon.props.color).toBe('#ffffff');
    });
  });
});

describe('AppbarContent', () => {
  (['small', 'medium', 'large', 'center-aligned'] as const).forEach((mode) =>
    it(`should render text component with appropriate variant for ${mode} mode`, () => {
      const { getByTestId } = render(
        <Appbar mode={mode}>
          <Appbar.Content title="Title" />
        </Appbar>
      );

      expect(getByTestId('appbar-content-title-text')).toHaveStyle(
        getTheme().fonts[modeTextVariant[mode]]
      );
    })
  );

  it('should render component passed to title', () => {
    const { getByText } = render(
      <Appbar>
        <Appbar.Content
          title={
            <Text testID="title" variant="displaySmall">
              Title
            </Text>
          }
        />
      </Appbar>
    );

    expect(getByText('Title')).toBeDefined();
  });
});

describe('getAppbarColors', () => {
  const elevation = 4;
  const customBackground = 'aquamarine';

  it('should return custom color no matter what is the theme version', () => {
    expect(getAppbarColor(getTheme(), elevation, customBackground)).toBe(
      customBackground
    );
  });

  it('should return v3 light color if theme version is 3', () => {
    expect(getAppbarColor(getTheme(), elevation)).toBe(
      tokens.md.ref.palette.neutral99
    );
  });

  it('should return v3 dark color if theme version is 3', () => {
    expect(getAppbarColor(getTheme(true), elevation)).toBe(
      tokens.md.ref.palette.neutral10
    );
  });

  it('should return v2 light color if theme version is 2', () => {
    expect(getAppbarColor(getTheme(false, false), elevation)).toBe('#6200ee');
  });

  it('should return v2 dark color if theme version is 2', () => {
    expect(getAppbarColor(getTheme(true, false), elevation)).toBe(
      overlay(elevation, '#121212')
    );
  });
});

describe('animated value changes correctly', () => {
  it('appbar animated value changes correctly', () => {
    const value = new Animated.Value(1);
    const { getByTestId } = render(
      <Appbar testID="appbar" style={[{ transform: [{ scale: value }] }]}>
        <Appbar.Action icon="menu" />
      </Appbar>
    );
    expect(getByTestId('appbar-outer-layer')).toHaveStyle({
      transform: [{ scale: 1 }],
    });

    Animated.timing(value, {
      toValue: 1.5,
      useNativeDriver: false,
      duration: 200,
    }).start();

    jest.advanceTimersByTime(200);

    expect(getByTestId('appbar-outer-layer')).toHaveStyle({
      transform: [{ scale: 1.5 }],
    });
  });

  it('action animated value changes correctly', () => {
    const value = new Animated.Value(1);
    const { getByTestId } = render(
      <Appbar>
        <Appbar.Action
          icon="menu"
          style={[{ transform: [{ scale: value }] }]}
          testID="appbar-action"
        />
      </Appbar>
    );
    expect(getByTestId('appbar-action-container-outer-layer')).toHaveStyle({
      transform: [{ scale: 1 }],
    });

    Animated.timing(value, {
      toValue: 1.5,
      useNativeDriver: false,
      duration: 200,
    }).start();

    jest.advanceTimersByTime(200);

    expect(getByTestId('appbar-action-container-outer-layer')).toHaveStyle({
      transform: [{ scale: 1.5 }],
    });
  });

  it('back action animated value changes correctly', () => {
    const value = new Animated.Value(1);
    const { getByTestId } = render(
      <Appbar>
        <Appbar.BackAction
          style={[{ transform: [{ scale: value }] }]}
          testID="appbar-back-action"
        />
      </Appbar>
    );
    expect(getByTestId('appbar-back-action-container-outer-layer')).toHaveStyle(
      {
        transform: [{ scale: 1 }],
      }
    );

    Animated.timing(value, {
      toValue: 1.5,
      useNativeDriver: false,
      duration: 200,
    }).start();

    jest.advanceTimersByTime(200);

    expect(getByTestId('appbar-back-action-container-outer-layer')).toHaveStyle(
      {
        transform: [{ scale: 1.5 }],
      }
    );
  });

  it('header animated value changes correctly', () => {
    const value = new Animated.Value(1);
    const { getByTestId } = render(
      <mockSafeAreaContext.SafeAreaProvider>
        <Appbar.Header
          style={[{ transform: [{ scale: value }] }]}
          testID="appbar-header"
        >
          {null}
        </Appbar.Header>
      </mockSafeAreaContext.SafeAreaProvider>
    );
    expect(getByTestId('appbar-header-outer-layer')).toHaveStyle({
      transform: [{ scale: 1 }],
    });

    Animated.timing(value, {
      toValue: 1.5,
      useNativeDriver: false,
      duration: 200,
    }).start();

    jest.advanceTimersByTime(200);

    expect(getByTestId('appbar-header-outer-layer')).toHaveStyle({
      transform: [{ scale: 1.5 }],
    });
  });
});
