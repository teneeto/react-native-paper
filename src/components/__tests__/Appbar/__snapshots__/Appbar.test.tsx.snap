// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Appbar does not pass any additional props to Searchbar 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "backgroundColor": "rgba(255, 251, 254, 1)",
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "height": 64,
      "left": undefined,
      "opacity": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "start": undefined,
      "top": undefined,
      "transform": undefined,
      "width": undefined,
    }
  }
  testID="surface-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": undefined,
        "paddingHorizontal": 4,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="surface"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": undefined,
          "backgroundColor": "rgb(238, 232, 244)",
          "borderRadius": 28,
          "bottom": undefined,
          "end": undefined,
          "flex": undefined,
          "height": undefined,
          "left": undefined,
          "opacity": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
          "transform": undefined,
          "width": undefined,
        }
      }
      testID="search-bar-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "rgb(238, 232, 244)",
            "borderRadius": 28,
            "flex": undefined,
            "flexDirection": "row",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="search-bar-container"
      >
        <View
          collapsable={false}
          style={
            Object {
              "alignSelf": undefined,
              "backgroundColor": "transparent",
              "borderRadius": 20,
              "bottom": undefined,
              "end": undefined,
              "flex": undefined,
              "height": 40,
              "left": undefined,
              "margin": 6,
              "opacity": undefined,
              "position": undefined,
              "right": undefined,
              "shadowColor": "#000",
              "shadowOffset": Object {
                "height": 0,
                "width": 0,
              },
              "shadowOpacity": 0,
              "shadowRadius": 0,
              "start": undefined,
              "top": undefined,
              "transform": undefined,
              "width": 40,
            }
          }
          testID="icon-button-container-outer-layer"
        >
          <View
            collapsable={false}
            style={
              Object {
                "backgroundColor": "transparent",
                "borderColor": "rgba(121, 116, 126, 1)",
                "borderRadius": 20,
                "borderWidth": 0,
                "elevation": 0,
                "flex": 1,
                "overflow": "hidden",
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
              }
            }
            testID="icon-button-container"
          >
            <View
              accessibilityComponentType="button"
              accessibilityLabel="search"
              accessibilityRole="button"
              accessibilityState={
                Object {
                  "disabled": true,
                }
              }
              accessibilityTraits="button"
              accessible={true}
              centered={true}
              collapsable={false}
              focusable={true}
              hitSlop={
                Object {
                  "bottom": 6,
                  "left": 6,
                  "right": 6,
                  "top": 6,
                }
              }
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                Array [
                  Object {
                    "overflow": "hidden",
                  },
                  false,
                  Array [
                    Object {
                      "alignItems": "center",
                      "flexGrow": 1,
                      "justifyContent": "center",
                    },
                    Object {
                      "borderRadius": 20,
                    },
                  ],
                ]
              }
              testID="icon-button"
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Û∞çâ
              </Text>
            </View>
          </View>
        </View>
        <TextInput
          accessibilityRole="search"
          keyboardAppearance="light"
          placeholder="Search"
          placeholderTextColor="rgba(28, 27, 31, 1)"
          returnKeyType="search"
          selectionColor="rgba(103, 80, 164, 1)"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "flex": 1,
                "fontSize": 18,
                "minWidth": 0,
                "paddingLeft": 8,
                "textAlign": "left",
              },
              Object {
                "color": "rgba(73, 69, 79, 1)",
                "fontFamily": "System",
                "fontSize": 16,
                "fontWeight": "400",
                "letterSpacing": 0.15,
                "lineHeight": 0,
              },
              Object {
                "minHeight": 56,
                "paddingLeft": 0,
              },
              undefined,
            ]
          }
          testID="search-bar"
          underlineColorAndroid="transparent"
          value=""
        />
        <View
          pointerEvents="none"
          style={
            Array [
              Object {
                "marginLeft": 16,
                "position": "absolute",
                "right": 0,
              },
              false,
            ]
          }
          testID="search-bar-icon-wrapper"
        >
          <View
            collapsable={false}
            style={
              Object {
                "alignSelf": undefined,
                "backgroundColor": "transparent",
                "borderRadius": 20,
                "bottom": undefined,
                "end": undefined,
                "flex": undefined,
                "height": 40,
                "left": undefined,
                "margin": 6,
                "opacity": undefined,
                "position": undefined,
                "right": undefined,
                "shadowColor": "#000",
                "shadowOffset": Object {
                  "height": 0,
                  "width": 0,
                },
                "shadowOpacity": 0,
                "shadowRadius": 0,
                "start": undefined,
                "top": undefined,
                "transform": undefined,
                "width": 40,
              }
            }
            testID="icon-button-container-outer-layer"
          >
            <View
              collapsable={false}
              style={
                Object {
                  "backgroundColor": "transparent",
                  "borderColor": "rgba(121, 116, 126, 1)",
                  "borderRadius": 20,
                  "borderWidth": 0,
                  "elevation": 0,
                  "flex": 1,
                  "overflow": "hidden",
                  "shadowColor": "#000",
                  "shadowOffset": Object {
                    "height": 0,
                    "width": 0,
                  },
                  "shadowOpacity": 0,
                  "shadowRadius": 0,
                }
              }
              testID="icon-button-container"
            >
              <View
                accessibilityComponentType="button"
                accessibilityLabel="clear"
                accessibilityRole="button"
                accessibilityState={
                  Object {
                    "disabled": false,
                  }
                }
                accessibilityTraits="button"
                accessible={true}
                centered={true}
                collapsable={false}
                focusable={true}
                hitSlop={
                  Object {
                    "bottom": 6,
                    "left": 6,
                    "right": 6,
                    "top": 6,
                  }
                }
                onBlur={[Function]}
                onClick={[Function]}
                onFocus={[Function]}
                onResponderGrant={[Function]}
                onResponderMove={[Function]}
                onResponderRelease={[Function]}
                onResponderTerminate={[Function]}
                onResponderTerminationRequest={[Function]}
                onStartShouldSetResponder={[Function]}
                style={
                  Array [
                    Object {
                      "overflow": "hidden",
                    },
                    false,
                    Array [
                      Object {
                        "alignItems": "center",
                        "flexGrow": 1,
                        "justifyContent": "center",
                      },
                      Object {
                        "borderRadius": 20,
                      },
                    ],
                  ]
                }
                testID="icon-button"
              >
                <Text
                  accessibilityElementsHidden={true}
                  allowFontScaling={false}
                  importantForAccessibility="no-hide-descendants"
                  pointerEvents="none"
                  selectable={false}
                  style={
                    Array [
                      Object {
                        "color": "rgba(255, 255, 255, 0)",
                        "fontSize": 24,
                      },
                      Array [
                        Object {
                          "lineHeight": 24,
                          "transform": Array [
                            Object {
                              "scaleX": 1,
                            },
                          ],
                        },
                        Object {
                          "backgroundColor": "transparent",
                        },
                      ],
                      Object {
                        "fontFamily": "Material Design Icons",
                        "fontStyle": "normal",
                        "fontWeight": "normal",
                      },
                      Object {},
                    ]
                  }
                >
                  Û∞Öñ
                </Text>
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;

exports[`Appbar passes additional props to AppbarBackAction, AppbarContent and AppbarAction 1`] = `
<View
  collapsable={false}
  style={
    Object {
      "alignSelf": undefined,
      "backgroundColor": "rgba(255, 251, 254, 1)",
      "bottom": undefined,
      "end": undefined,
      "flex": undefined,
      "height": 64,
      "left": undefined,
      "opacity": undefined,
      "position": undefined,
      "right": undefined,
      "shadowColor": "#000",
      "shadowOffset": Object {
        "height": 0,
        "width": 0,
      },
      "shadowOpacity": 0,
      "shadowRadius": 0,
      "start": undefined,
      "top": undefined,
      "transform": undefined,
      "width": undefined,
    }
  }
  testID="surface-outer-layer"
>
  <View
    collapsable={false}
    style={
      Object {
        "alignItems": "center",
        "backgroundColor": "rgba(255, 251, 254, 1)",
        "flex": 1,
        "flexDirection": "row",
        "paddingBottom": undefined,
        "paddingHorizontal": 4,
        "paddingLeft": undefined,
        "paddingRight": undefined,
        "paddingTop": undefined,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 0,
          "width": 0,
        },
        "shadowOpacity": 0,
        "shadowRadius": 0,
      }
    }
    testID="surface"
  >
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": undefined,
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "bottom": undefined,
          "end": undefined,
          "flex": undefined,
          "height": 40,
          "left": undefined,
          "margin": 6,
          "opacity": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
          "transform": undefined,
          "width": 40,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityLabel="Back"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessibilityTraits="button"
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 20,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <View
                style={
                  Array [
                    Object {
                      "alignItems": "center",
                      "justifyContent": "center",
                    },
                    Object {
                      "height": 24,
                      "transform": Array [
                        Object {
                          "scaleX": 1,
                        },
                      ],
                      "width": 24,
                    },
                  ]
                }
              >
                <Image
                  accessibilityIgnoresInvertColors={true}
                  source={
                    Object {
                      "testUri": "../../../src/assets/back-chevron.png",
                    }
                  }
                  style={
                    Array [
                      Object {
                        "resizeMode": "contain",
                      },
                      Object {
                        "height": 21,
                        "tintColor": "rgba(28, 27, 31, 1)",
                        "width": 21,
                      },
                    ]
                  }
                />
              </View>
            </View>
          </View>
        </View>
      </View>
    </View>
    <View
      accessibilityRole="text"
      accessibilityState={
        Object {
          "disabled": true,
        }
      }
      accessible={true}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      pointerEvents="box-none"
      style={
        Array [
          Object {
            "flex": 1,
            "paddingHorizontal": 12,
          },
          Object {
            "paddingHorizontal": 0,
          },
          Array [
            false,
            false,
            undefined,
          ],
        ]
      }
      testID="appbar-content"
    >
      <Text
        accessibilityRole="header"
        accessible={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "fontFamily": "System",
              "fontSize": 22,
              "fontWeight": "400",
              "letterSpacing": 0,
              "lineHeight": 28,
            },
            Object {
              "textAlign": "left",
            },
            Object {
              "color": "rgba(28, 27, 31, 1)",
              "writingDirection": "ltr",
            },
            Array [
              Object {
                "color": "rgba(28, 27, 31, 1)",
                "fontFamily": "System",
                "fontSize": 22,
                "fontWeight": "400",
                "letterSpacing": 0,
                "lineHeight": 28,
              },
              false,
              undefined,
            ],
          ]
        }
        testID="appbar-content-title-text"
      >
        Examples
      </Text>
    </View>
    <View
      collapsable={false}
      style={
        Object {
          "alignSelf": undefined,
          "backgroundColor": "transparent",
          "borderRadius": 20,
          "bottom": undefined,
          "end": undefined,
          "flex": undefined,
          "height": 40,
          "left": undefined,
          "margin": 6,
          "opacity": undefined,
          "position": undefined,
          "right": undefined,
          "shadowColor": "#000",
          "shadowOffset": Object {
            "height": 0,
            "width": 0,
          },
          "shadowOpacity": 0,
          "shadowRadius": 0,
          "start": undefined,
          "top": undefined,
          "transform": undefined,
          "width": 40,
        }
      }
      testID="icon-button-container-outer-layer"
    >
      <View
        collapsable={false}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "rgba(121, 116, 126, 1)",
            "borderRadius": 20,
            "borderWidth": 0,
            "elevation": 0,
            "flex": 1,
            "overflow": "hidden",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 0,
              "width": 0,
            },
            "shadowOpacity": 0,
            "shadowRadius": 0,
          }
        }
        testID="icon-button-container"
      >
        <View
          accessibilityComponentType="button"
          accessibilityRole="button"
          accessibilityState={
            Object {
              "disabled": false,
            }
          }
          accessibilityTraits="button"
          accessible={true}
          centered={true}
          collapsable={false}
          focusable={true}
          hitSlop={
            Object {
              "bottom": 6,
              "left": 6,
              "right": 6,
              "top": 6,
            }
          }
          onBlur={[Function]}
          onClick={[Function]}
          onFocus={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Array [
              Object {
                "overflow": "hidden",
              },
              false,
              Array [
                Object {
                  "alignItems": "center",
                  "flexGrow": 1,
                  "justifyContent": "center",
                },
                Object {
                  "borderRadius": 20,
                },
              ],
            ]
          }
          testID="icon-button"
        >
          <View
            style={
              Array [
                Object {
                  "alignItems": "center",
                  "justifyContent": "center",
                },
                Object {
                  "height": 24,
                  "width": 24,
                },
              ]
            }
          >
            <View
              collapsable={false}
              style={
                Object {
                  "bottom": 0,
                  "left": 0,
                  "opacity": 1,
                  "position": "absolute",
                  "right": 0,
                  "top": 0,
                  "transform": Array [
                    Object {
                      "rotate": "0deg",
                    },
                  ],
                }
              }
            >
              <Text
                accessibilityElementsHidden={true}
                allowFontScaling={false}
                importantForAccessibility="no-hide-descendants"
                pointerEvents="none"
                selectable={false}
                style={
                  Array [
                    Object {
                      "color": "rgba(73, 69, 79, 1)",
                      "fontSize": 24,
                    },
                    Array [
                      Object {
                        "lineHeight": 24,
                        "transform": Array [
                          Object {
                            "scaleX": 1,
                          },
                        ],
                      },
                      Object {
                        "backgroundColor": "transparent",
                      },
                    ],
                    Object {
                      "fontFamily": "Material Design Icons",
                      "fontStyle": "normal",
                      "fontWeight": "normal",
                    },
                    Object {},
                  ]
                }
              >
                Û∞çú
              </Text>
            </View>
          </View>
        </View>
      </View>
    </View>
  </View>
</View>
`;
